src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/client/CommunityApiClient.kt

```diff
@@ -28,14 +29,24 @@ class CommunityApiClient(private val webClient: WebClient) {
     private val log = LoggerFactory.getLogger(this::class.java)
   }
 
-  fun getActiveConvictions(crn: String): Mono<List<Conviction>> {
-    val responseType = object : ParameterizedTypeReference<List<Conviction>>() {}
+  fun getActiveConvictions(crn: String): Flux<Conviction> {
+    val responseType = object : ParameterizedTypeReference<Conviction>() {}
 
     return webClient
       .get()
       .uri("/offenders/crn/$crn/convictions?activeOnly=true")
       .retrieve()
-      .bodyToMono(responseType)
+      .onStatus(
+        { httpStatus -> HttpStatus.NOT_FOUND == httpStatus },
+        { Mono.error(MissingConvictionError("No active convictions found for $crn")) }
+      )
+      .bodyToFlux(responseType)
+      .onErrorResume { ex ->
+        when (ex) {
+          is MissingConvictionError -> Mono.empty()
```

Comment:
I think Flux have similar methods
```suggestion
          is MissingConvictionError -> Flux.empty()
```